src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/PhysicalAddress.cs

```diff
@@ -103,22 +103,44 @@ public byte[] GetAddressBytes()
             return (byte[])_address.Clone();
         }
 
-        public static PhysicalAddress Parse(string address)
+        public static PhysicalAddress Parse(string address) => address != null ? Parse(address.AsSpan()) : None;
+
+        public static PhysicalAddress Parse(ReadOnlySpan<char> address)
         {
-            int validSegmentLength;
-            char? delimiter = null;
-            byte[] buffer;
+            if (!TryParseInternal(address, out PhysicalAddress value))
+                ThrowBadAddressException(address);
 
+            return value;
+
+            static void ThrowBadAddressException(ReadOnlySpan<char> address) =>
+                throw new FormatException(SR.Format(SR.net_bad_mac_address, new string(address)));
```

Comment:
Why separate this out into a separate throw method?  Are you concerned about the Parse being inlineable and this preventing that, or something else?  We generally only do this when a) we're going to do the throw in multiple places, b) we're using it in generic code and we want to avoid multiple instantiations, and/or c) we're trying to keep the call site as lean as possible to be inlineable for perf reasons.  (a) and (b) don't apply here.  Consider keeping it simple:
```C#
if (!TryParseInternal(address, out PhysicalAddress result))
{
    throw new FormatException(SR.Format(SR.net_bad_mac_address, new string(address)));
}

return value;
```

Reply:
It's a leftover from the original version where exception was thrown in several places. I will simplify it as you suggested.